### ISSUE TITLE
App no longer works with file in Dropbox app

### ISSUE THREAD

**Issue by RKopacz** [bpellin/keepassdroid#414]
I have been using this app for (I think) almost a decade now. I have been keeping my database file in Dropbox. When I clicked on the file name in the Dropbox app, it would auto suggest this app, I would select it, and it would open.

For the past several weeks now, when I click on the database file, it instead auto suggests that I open it as a PDF, and does not give me the option of the Keepass droid app.

I have tried downloading the file to my phone and testing it there, but it still exhibits the new behavior, so I am guessing it has nothing to do with any changes to the Dropbox app.

The database file works with the desktop version of Keepass

Is this a known issue, and is there a fix in progress?

**Comment by DrRob** [comment]
I have this problem too.  There is a workaround - open the Dropbox menu on the file and toggle "Make available offline" off and on again, and then when you tap the file it will ask which app to use, select Keepassdroid and it will open.  Unfortunately the workaround doesn't last for long and has to be repeated.

I've switched to KeePassDX, which doesn't have this issue.

**Comment by DrRob** [comment]
I spoke too soon - I'm seeing the same problem with KeePassDX now.

**Comment by J-Jamet** [comment]
The answer is very simple, it is a regression in Dropbox. They must therefore be contacted directly, so good luck! :zany_face:

**Comment by DrRob** [comment]
Yep, I've spoken with Dropbox support about this at length, and we tried many things but they ran out of ideas.  I've posted about it in the Dropbox community forums now, and Support has passed it on to Development too.

**Comment by RKopacz** [comment]
If it is a Dropbox issue, then why does the phone still not auto-suggest this app after the file has been downloaded to the phone's file system? I've tested that and it still won't auto-suggest the app. Either the API for Android for this particular operation has changed, or something in the app has changed.

**Comment by nhalink** [comment]
I had the same problem, went looking and found this:
https://www.dropboxforum.com/t5/Dropbox-files-folders/Android-Kdbx-files-can-only-opened-with-Google-Drive-PDF-Viewer/m-p/427607/highlight/true#M163540

Dropbox ditched the native program-picker and implemented one of their own.
This program-picker can only attach a certain app to an extension when the app has registered that extension.

In my opinion this is way too strict, because I'm sure a lot of apps don't have this optional setting.

But either dropbox will have too loosen this registered extension with app or keepassdroid has to register the extensions.

**Comment by dave1812** [comment]
My wife and I recently can't access KEEPASS from DROPBOX.  I tried reinstalling both and deleting data.  No joy.  I think the issue is on Dropbox's end because this happened within the last couple of years--same issue.  and then they released a new version of dropbox.

**Comment by DrRob** [comment]
I've discussed this at length with Dropbox support and the bottom line is that Dropbox doesn't support all types of what they refer to as "database" files and they have no plan to fix this.

**Comment by antivirtel** [comment]
Thanks for the update, I think it's time to migrate to Google Drive. I didn't migrate so far since I got the problem with no desktop client on Linux, but Android is way more important.

**Comment by dave1812** [comment]
I found a workaround for my issue.  I didn't realize that a recent Dropbox update removed the setting for my Keepass database file to be available offline.   As soon as I noticed it was no longer set to be offline, I set it offline and presto, Keepass loaded when I clicked on my database file.   Regardless, Dropbox is not working correctly, that the file must be set to offline in order to work with Keepass.

**Comment by antivirtel** [comment]
No, it doesn't work for me. Still the same issue: if you edit the file, it uploads it back to Dropbox, but if you try to open it up again (password or fingerprint), "Invalid password or keyfile". So go to Dropbox website (Desktop), find the file, restore previous version. Only edit on PC/Mac/Linux with KeePassXC. :( I'd say time to move to Google Drive, there's a FOSS Linux client too, hopefully that works
